tests/lib/rules/component-tags-order.js

```diff
@@ -268,7 +345,23 @@ tester.run('component-tags-order', rule, {
           message: 'The <script> should be above the <style> on line 2.',
           line: 3
         }
-      ]
+      ],
+      output: '\n        <script></script>\n        <style></style>\n      '
     }
   ]
 })
+
+describe('suppress reporting with eslint-disable-next-line', () => {
+  it('do not report if <!-- eslint-disable-next-line -->', async () => {
+    const code = `<style></style><template></template>
+    <!-- eslint-disable-next-line vue/component-tags-order -->
+    <script></script>`
+    const [{ messages }] = await eslint.lintText(code, { filePath: 'test.vue' })
+    assert.strictEqual(messages.length, 1)
+    // should not fix <script>
+    assert.strictEqual(
+      messages[0].fix.text,
+      '<template></template><style></style>'
+    )
```

Comment:
I think it's better to check the fixed text using the `output` provided by the ESLint class. To use the `output`, add `fix: true` to the ESLint constructor option.

```suggestion
    const [{ messages, output }] = await eslint.lintText(code, { filePath: 'test.vue' })
    assert.strictEqual(messages.length, 0)
    // should not fix <script>
    assert.strictEqual(
      output,
      `<template></template><style></style>
    <!-- eslint-disable-next-line vue/component-tags-order -->
    <script></script>`
    )
```